lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp

```diff
@@ -61,6 +61,26 @@ std::vector<c10::ScalarType> compute_dtype_embedding_dense_backward(const at::Te
   return {grad_output.scalar_type()};
 }
 
+std::vector<std::vector<int64_t>> compute_shape_index_select(const at::Tensor & self, int64_t dim,
+    const at::Tensor & index) {
+  // Based on definition of https://pytorch.org/docs/stable/generated/torch.index_select.html.
+  // Promote Rank 0 index tensor to a 1 * 1 tensor.
+  auto index_dim = index.dim() > 0 ? index.dim() : 1;
```

Comment:
when can index.dim() be 0?  I found that it is invalid to use this:

```
>>> torch.index_select(x, -1, torch.tensor())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: tensor() missing 1 required positional arguments: "data"
>>> torch.index_select(x, -1, torch.tensor([]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: index_select(): Expected dtype int32 or int64 for index
>>> torch.index_select(x, -1, torch.tensor([0]))
tensor([[1.],
        [0.],
        [0.],
        [0.],
        [0.]])
```

Reply:
We have this test case `AtenLtcTsTensorTest.TestIndexSelectRank0` which seems passing all the time... Not sure...